CRM/Report/Form/Contact/Relationship.php

```diff
@@ -774,4 +783,48 @@ public function buildValidityQuery($valid) {
     return $clause;
   }
 
+  /**
+   * Get SQL where clause for a active period field.
+   *
+   * @param string $fieldName
+   * @param string $relative
+   * @param string $from
+   * @param string $to
+   * @param string $type
+   * @param string $fromTime
+   * @param string $toTime
+   *
+   * @return null|string
+   */
+  public function activeClause(
+    $fieldName,
+    $relative, $from, $to, $type = NULL, $fromTime = NULL, $toTime = NULL
+    ) {
+    $clauses = array();
+    if (in_array($relative, array_keys($this->getOperationPair(CRM_Report_Form::OP_DATE)))) {
+      return NULL;
+    }
+
+    list($from, $to) = $this->getFromTo($relative, $from, $to, $fromTime, $toTime);
+
+    if ($from) {
+      $from = ($type == CRM_Utils_Type::T_DATE) ? substr($from, 0, 8) : $from;
+    }
+
+    if ($to) {
+      $to = ($type == CRM_Utils_Type::T_DATE) ? substr($to, 0, 8) : $to;
+    }
+
+    if ($from && $to) {
+      return CRM_Contact_BAO_Query::getRelationshipActivePeriodClauses($from, $to, FALSE);
+    }
+    elseif ($from) {
+      return CRM_Contact_BAO_Query::getRelationshipActivePeriodClauses($from, NULL, FALSE);
+    }
+    elseif ($to) {
+      return CRM_Contact_BAO_Query::getRelationshipActivePeriodClauses(NULL, $to, FALSE);
+    }
```

Comment:
Can we remove the additional `elseif()`(line 821 - 826) here with `&&` replaced by `||` in line 818 ? I think `getRelationshipActivePeriodClauses()` should take care of proper return clause as the check for $from & $to is done there too.